response.go

```diff
@@ -7,18 +7,39 @@ import (
 	"github.com/vmihailenco/msgpack/v5"
 )
 
-type Response struct {
-	RequestId uint32
-	Code      uint32
-	// Error contains an error message.
-	Error string
-	// Data contains deserialized data for untyped requests.
-	Data []interface{}
-	// Pos contains a position descriptor of last selected tuple.
-	Pos      []byte
-	MetaData []ColumnMetaData
-	SQLInfo  SQLInfo
-	buf      smallBuf
+type Response interface {
+	Decode() ([]interface{}, error)
+	DecodeTyped(res interface{}) error
+	SetBuf(buf interface{}) error
+}
+
+type BaseResponse struct {
```

Comment:
Does it could be a private?
```suggestion
type baseResponse struct {
```